### ISSUE TITLE
OpenMM failure to load .prmtop file

### ISSUE THREAD

**Issue by apeck12** [pandegroup/openmm#887]
A prmtop file generated in Amber/CHARMM36 is failing to load in OpenMM. Specifically, it cannot recognize the following line: 
`%FORMAT(i2,a78)`
and returns Nonetype as a result.

The example .prmtop file is available here: https://gist.github.com/apeck12/c32c3760ff1fbae989da

The error message is below:

```
In [1]: from simtk.openmm.app import AmberPrmtopFile 

In [2]: AmberPrmtopFile('apocam_seed0.prmtop')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-1040e2410961> in <module>()
----> 1 AmberPrmtopFile('apocam_seed0.prmtop')

/home/rmcgibbo/miniconda/envs/3.4.2/lib/python3.4/site-packages/simtk/openmm/app/amberprmtopfile.py in __init__(self, file)
     81         # Load the prmtop file
     82 
---> 83         prmtop = amber_file_parser.PrmtopLoader(file)
     84         self._prmtop = prmtop
     85 

/home/rmcgibbo/miniconda/envs/3.4.2/lib/python3.4/site-packages/simtk/openmm/app/internal/amber_file_parser.py in __init__(self, inFilename)
    135                     format = format[index0+1:index1]
    136                     m = FORMAT_RE_PATTERN.search(format)
--> 137                     self._raw_format[self._flags[-1]] = (format, m.group(1), m.group(2), m.group(3), m.group(4))
    138                 elif self._flags \
    139                      and 'TITLE'==self._flags[-1] \

AttributeError: 'NoneType' object has no attribute 'group'
```

Thanks.

**Comment by swails** [comment]
OpenMM does not support chamber-style topology files (and its Fortran Format regex is too simplistic to handle some of the Fortran format specs used in the chamber prmtop file format).

You can try [ParmEd](http://github.com/ParmEd/ParmEd) instead, which has much more complete Amber support (i.e., 10-12 nonbonded support, old-style prmtop file support, and chamber prmtop support).  The beginning of your script should look something like this:

```Python
from chemistry.amber import LoadParm
from simtk.unit import *
from simtk.openmm import *
from simtk.openmm.app import *

# It's best to supply both a prmtop and inpcrd to set the default periodic box vectors
# correctly if there are any
parm = LoadParm('apocam_seed0.prmtop', 'apocam_seed0.inpcrd')

# The call signature is basically the same as AmberPrmtopFile
system = parm.createSystem(nonbondedMethod=PME, constraints=HBonds,
                           nonbondedCutoff=1*nanometers)

# The rest is the same as with OpenMM
```

ParmEd has some other stuff as well, including a NetCDF reporter (so you can write Amber-style NetCDF trajectories instead of DCDs or PDBs), all of which is described in the "OpenMM" section of the ParmEd doc webpage (http://parmed.github.io/ParmEd).

Of course, there's a rather simple alternative to all of this -- use the CHARMM files directly in OpenMM instead.  (ParmEd also supports using CHARMM PSF files to go straight to OpenMM -- see [here](http://parmed.github.io/ParmEd/html/openmm.html) for examples and documentation).

**Comment by rmcgibbo** [comment]
If we're not going to support this, we should at least give error messages as opposed to crazy tracebacks.

**Comment by rmcgibbo** [comment]
The existing code is clearly buggy in the sense that it uses regular expression capture groups without even checking to make sure that the expression matched.

**Comment by swails** [comment]
Arbitrary Fortran format parsing is nontrivial at best (see the fortranformat project on bitbucket).  Unfortunately, the fortranformat parser is painfully slow (it would cause a 10x slowdown in the existing parser) -- the code in ParmEd to properly handle arbitrary Fortran formats in an efficient way is beyond what I'm willing to spend the time to port back.

I can add a sane error message to the existing parser, and punt users to the CHARMM parsers or ParmEd instead.

**Comment by swails** [comment]
FWIW, I tried to add an error message in #764.  My mistake there was to assume that the prmtop itself would parse correctly, and that only when trying to create a system should it throw an exception.

**Comment by swails** [comment]
I will submit a new PR that gives a comprehensible error message within a week.

**Comment by swails** [comment]
This was fixed in #892 and can be closed.
